compat/mingw.c

```diff
@@ -3718,9 +3722,13 @@ static void setup_windows_environment(void)
 		char buf[32768];
 		size_t off = 0;
 
-		xsnprintf(buf, sizeof(buf),
-			  "MINGW%d", (int)(sizeof(void *) * 8));
-		setenv("MSYSTEM", buf, 1);
+#if defined(__MINGW64__) || defined(_M_AMD64)
+		setenv("MSYSTEM", "MINGW64", 1);
+#elif defined(__aarch64__) || defined(_M_ARM64) || defined(_M_ARM64EC)
```

Comment:
I don't think our clang can do `ARM64EC`, and I'm not sure if that would even be the same `MSYSTEM`, but i guess `CLANGARM64` is the closest thing to the correct `MSYSTEM` if anyone where to compile for `ARM64EC`

Reply:
I'm glad we agree!